deployment/recipes/library/volttron_agent.py

```diff
@@ -0,0 +1,633 @@
+# Battelle Memorial Institute. The views and opinions of authors expressed
+# herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY operated by
+# BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+# }}}
+
+
+# Note for reference, this module is developed per the patter in the ansible
+# docs here: https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html
+
+import csv
+import io
+import glob
+import json
+import os
+import subprocess
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: volttron_agents
+
+short_description: Manage the state of agents in a platform
+
+description:
+    - Manage the lifecycle and configuration state of agents installed in a platform.
+    - Note, thorough inspection of agent configuration and state is subtle; this module is
+      intended to simply fail with a useful message in those cases. It may be necessary to
+      remove and re-install agents in these situations.
+    - Note, this module assumes that there is a running platform available for it to interact with.
+
+options:
+    volttron_root:
+        description:
+            - path to the VOLTTRON source tree (where start_volttron and stop_volttron are found)
+        default: $HOME/volttron
+        type: path
+    volttron_venv:
+        description:
+            - path to the virtual environment where VOLTTRON is installed
+        default: $volttron_root/env
+        type: path
+    volttron_home:
+        description:
+            - path to the VOLTTRON_HOME directory for this instance
+        default: $HOME/.volttron'
+        type: path
+    agent_configs_dir:
+        description:
+            - path to directory on the remote system where agent configuration files have been placed
+        default: $HOME/configs
+        type: path
+    time_limit:
+        description:
+            - Time limit, in seconds, for subprocess calls used by the module.
+        type: int
+        default: 30
+    agent_vip_id:
+        description:
+            - vip identity of the agent to be installed
+        type: string
+        required: true
+    agent_spec:
+        description:
+            - a dictionary of configuration details for the agent
+            - valid keys are:
+                - agent_state: (either 'present' or 'absent'), indicates if the agent should or should not be present
+                - agent_enabled: (bool), if true, indicates that the agent should be started when the platform starts (does not imply running)
+                - agent_priority: (int; default 50) sets the priority ordering for starting an enalbed agent
+                - agent_running: (bool), if true, indicates that the agent should be started as it is installed (does not imply enabled)
```

Comment:
I am not sure I like the agent_running paramenter.  Would agent_start work better here?  It seems like agent_running would be a state of the agent?